numpy/core/src/umath/ufunc_object.c

```diff
@@ -3520,6 +3524,10 @@ reduce_loop(NpyIter *iter, char **dataptrs, npy_intp const *strides,
             innerloop(dataptrs_copy, &count,
                         strides_copy, innerloopdata);
 
+            if (needs_api && PyErr_Occurred()) {
+                break;
```

Comment:
This break needs to be a `goto finish` or so. Pandas is running into this in their CI.  Will look into it tomorrow at the latest, but the only tricky thing is that it might be nice to have a test.

Reply:
I'm going to check it out now, unless you already have it done

Reply:
Thanks, I thought a test would be more complicated for some reason.  I think the `iternext` technically also should be checked, although I am willing to bet is not a regression.